src/fileoverview_comment_transformer.ts

```diff
@@ -61,89 +61,106 @@ function augmentFileoverviewComments(tags: jsdoc.Tag[]) {
+        // fileoverview comment.
+        for (let i = originalComments.length - 1; i >= 0; i--) {
+          const end = originalComments[i].end;
+          if (!text.substring(end).startsWith('\n\n') &&
+              !text.substring(end).startsWith('\r\n\r\n')) {
+            continue;
           }
+          // This comment is separated from the source file with a double break, marking it (and any
+          // preceding comments) as a file-level comment. Split them off and attach them onto a
+          // NotEmittedStatement, so that they do not get lost later on.
+          const synthesizedComments = jsdoc.synthesizeLeadingComments(firstStatement);
+          const notEmitted = ts.createNotEmittedStatement(sourceFile);
+          // Modify the comments on the firstStatement in place by removing the file-level comments.
+          fileComments = synthesizedComments.splice(0, i + 1);
+          // Move the fileComments onto notEmitted.
+          ts.setSyntheticLeadingComments(notEmitted, fileComments);
+          sourceFile = updateSourceFileNode(
+              sourceFile,
+              ts.createNodeArray([notEmitted, firstStatement, ...sourceFile.statements.slice(1)]));
+          break;
+        }
+
+
+        // Now walk every top level statement and escape/drop any @fileoverview comments found.
+        // Closure ignores all @fileoverview comments but the last, so tsickle must make sure not to
+        // emit duplicated ones.
+        for (let i = 0; i < sourceFile.statements.length; i++) {
+          const stmt = sourceFile.statements[i];
+          // Accept the NotEmittedStatement inserted above.
+          if (i === 0 && stmt.kind === ts.SyntaxKind.NotEmittedStatement) continue;
+          const comments = jsdoc.synthesizeLeadingComments(stmt);
+          checkNoFileoverviewComments(
+              stmt, comments,
+              `file comments must be at the top of the file, ` +
+                  `separated from the file body by an empty line.`);
         }
       }
-    }
 
-    // Closure Compiler considers the *last* comment with @fileoverview (or @externs or @nocompile)
-    // that has not been attached to some other tree node to be the file overview comment, and
-    // only applies @suppress tags from it.
-    // AJD considers *any* comment mentioning @fileoverview.
-    let fileoverviewIdx = -1;
-    let tags: jsdoc.Tag[] = [];
-    for (let i = fileComments.length - 1; i >= 0; i--) {
-      const parse = jsdoc.parseContents(fileComments[i].text);
-      if (parse !== null && parse.tags.some(t => FILEOVERVIEW_COMMENT_MARKERS.has(t.tagName))) {
-        fileoverviewIdx = i;
-        tags = parse.tags;
-        break;
+      // Closure Compiler considers the *last* comment with @fileoverview (or @externs or
+      // @nocompile) that has not been attached to some other tree node to be the file overview
+      // comment, and only applies @suppress tags from it. AJD considers *any* comment mentioning
```

Comment:
Is AJD public?  Maybe just add "(a Google-internal tool)" if not.

Reply:
Done.